File: cosmos/Cosmos-API/src/main/java/com/planbtech/cosmos/model/entites/MaritalStatus.java
package com.planbtech.cosmos.model.entites;

import com.planbtech.cosmos.model.Interfaces.IMartialStatus;
import jakarta.persistence.*;

@Entity(name = "maritalstatus")
public class MaritalStatus implements IMartialStatus {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "maritalstatus_id")
    private Long id;

    @Column(name = "maritalstatus")
    private String maritalStatus;

    @Override
    public Long getId() {
        return this.id;
    }

    @Override
    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public String getMaritalStatus() {
        return this.maritalStatus;
    }

    @Override
    public void setMaritalStatus(String matrialStatus) {
        this.maritalStatus = matrialStatus;
    }
}

File: cosmos/ProConnect/src/app/Class/maritalstatus.ts
import { Imaritalstatus } from "../interfaces/imaritalstatus";

export class Maritalstatus implements Imaritalstatus{
  maritalStatus: string = '';
  id: number = 0;


  getMaritalStatus():string{return this.maritalStatus}
  getId():number{return this.id}

  setMaritalStatus(maritalStatus: string): void {this.maritalStatus = maritalStatus  }
  setId(id: number): void {this.id = id  }
}
